andela/balder-ah-backend
# #162170712 Create users profiles

#### What does this PR do?
This PR handles retrieving a logged in or signed up user's profile, editing a user's profile and getting other user's profile
#### Description of Task to be completed?
This task has the following endpoints working

- GET /api/user/
To retrieve a current user's profile/details
- PUT /api/user/
To update a current user's profile/details
- GET /api/profiles/:username
To get other user's profile

#### How should this be manually tested?
- Clone this repo
- In your terminal cd into the folder it creates
- Run **npm install** to install all the dependencies that will be used by the project
- Check the **.env.sample** to create a **.env** file in your root folder
- Run **npm run dev:start** to start the app
- Signup through **localhost:3000/api/users/signup** or Login through **localhost:3000/api/users/login** respectively using postman as a client
- Send a GET request to **localhost:3000/api/user** to get your profile details
- Send a PUT request to **localhost:3000/api/user** with the required fields (username, email, bio, image) filled to update your profile details
- Send a GET request to **localhost:3000/api/profiles/(enter username)** to retrieve a user's profile
- Run **npm test** to test the app
- View the Swagger UI on **localhost:3000/api/docs**
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#162170712](https://www.pivotaltracker.com/story/show/162170712)
#### Screenshots (if appropriate)
<img width="1440" alt="screen shot 2018-12-04 at 4 19 54 pm" src="https://user-images.githubusercontent.com/29441898/49451991-a4895a00-f7e0-11e8-98ac-cb8bee295a29.png">
<img width="1440" alt="screen shot 2018-12-04 at 4 20 15 pm" src="https://user-images.githubusercontent.com/29441898/49451992-a4895a00-f7e0-11e8-8b0c-e585d1268f5d.png">
<img width="1440" alt="screen shot 2018-12-04 at 4 20 18 pm" src="https://user-images.githubusercontent.com/29441898/49451993-a4895a00-f7e0-11e8-8746-3bb5c12e11d9.png">

#### Questions:
